pkg/state/state_exec_tmpl.go

```diff
@@ -94,6 +90,9 @@ func (st *HelmState) ExecuteTemplates() (*HelmState, error) {
 	vals := st.Values()
 
 	for i, rt := range st.Releases {
+		if rt.KubeContext == "" {
+			rt.KubeContext = r.HelmDefaults.KubeContext
+		}
```

Comment:
Ah, nice. This looks like the right place to do defaulting. I might rename `ExecuteTemplates` later to reflect that... perhaps `ApplyDefaultsAndReleaseTemplates` 🤣